### PR TITLE
Disable immutable inputs when editing a challenge

### LINKED ISSUE #9394: Shouldn't change short name/gem prize
The Challenge edit form should not allow the short name (tag name) or gem prize to be changed.

### PR DESCRIPTION
**Pull request by librarianmage**
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

[//]: # (Put Issue # or URL here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes #9394 

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)

[//]: # (Put User ID in here - found on the Habitica website at User Icon > Settings > API)

----
UUID:

### CODE REVIEW

File: website/client/components/challenges/challengeModal.vue [hunk 1]
```diff
@@ -47,10 +47,10 @@
         label
           strong(v-once) {{$t('endDate')}}
         b-form-input.end-date-input
-      .form-group(v-if='creating')
+      .form-group
```
Comment (paglias) [inline website/client/components/challenges/challengeModal.vue]: @MathWhiz this part of the form shouldn't be shown at all while a challenge is being edited, why didn't the v-if work already?

**Review by paglias — CHANGES_REQUESTED** [review]

File: website/client/components/challenges/challengeModal.vue (hunk 1 above)
Comment (librarianmage) [reply website/client/components/challenges/challengeModal.vue]: Oh, I was just testing to see if the :disabled property worked, and forgot to reset it.

**Comment by librarianmage** [comment]
@paglias Updated!

**Review by paglias — APPROVED** [review]

**Comment by librarianmage** [comment]
Hello! It appears that this hasn't been added to my entry in the Hall of heroes. Is that intentional? My user id is 95f16006-1504-4a9f-95b6-0c7314dbd7ee.

**Comment by Alys** [comment]
@MathWhiz Fixed! Thanks again!

**Comment by librarianmage** [comment]
:D thank you for adding it!

**Comment by TheHollidayInn** [comment]
Sorry, that's my bad!

**Pull request merged**